src/unit/test_hashtable.c

```diff
@@ -945,3 +945,315 @@ int test_random_entry_sparse_table(int argc, char **argv, int flags) {
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+
+    /* Test safe iterator with NULL hashtable */
+    hashtableIterator safe_iter;
+    hashtableInitIterator(&safe_iter, NULL, HASHTABLE_ITER_SAFE);
+
+    /* hashtableNext should return false for NULL hashtable */
+    void *entry;
+    TEST_ASSERT(!hashtableNext(&safe_iter, &entry));
+
+    /* Reset should handle NULL hashtable gracefully */
+    hashtableCleanupIterator(&safe_iter);
+
+    /* Test non-safe iterator with NULL hashtable */
+    hashtableIterator unsafe_iter;
+    hashtableInitIterator(&unsafe_iter, NULL, 0);
+
+    /* hashtableNext should return false for NULL hashtable */
+    TEST_ASSERT(!hashtableNext(&unsafe_iter, &entry));
+
+    /* Reset should handle NULL hashtable gracefully */
+    hashtableCleanupIterator(&unsafe_iter);
+
+    /* Test reinitializing NULL iterator with valid hashtable */
+    hashtableType type = {0};
+    hashtable *ht = hashtableCreate(&type);
+    TEST_ASSERT(hashtableAdd(ht, (void *)1));
+
+    hashtableRetargetIterator(&safe_iter, ht);
+    TEST_ASSERT(hashtableNext(&safe_iter, &entry));
+    TEST_ASSERT(entry == (void *)1);
+
+    hashtableCleanupIterator(&safe_iter);
+    hashtableRelease(ht);
+
+    return 0;
+}
+
+int test_hashtable_retarget_iterator(int argc, char **argv, int flags) {
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+
+    hashtableType type = {0};
+    hashtable *ht1 = hashtableCreate(&type);
+    hashtable *ht2 = hashtableCreate(&type);
+
+    /* Add different entries to each hashtable */
+    for (int i = 0; i < 5; i++) {
+        TEST_ASSERT(hashtableAdd(ht1, (void *)(long)(i + 10)));
+        TEST_ASSERT(hashtableAdd(ht2, (void *)(long)(i + 20)));
+    }
+
+    /* Create iterator on first hashtable */
+    hashtableIterator iter;
+    hashtableInitIterator(&iter, ht1, HASHTABLE_ITER_SAFE);
+
+    /* Iterate partially through first hashtable */
+    void *entry;
+    int count1 = 0;
+    while (hashtableNext(&iter, &entry) && count1 < 3) {
+        long val = (long)entry;
+        TEST_ASSERT(val >= 10 && val < 15);
+        count1++;
+    }
+
+    /* Retarget to second hashtable */
+    hashtableRetargetIterator(&iter, ht2);
+
+    /* Iterate partially through second hashtable */
+    int count2 = 0;
+    while (hashtableNext(&iter, &entry) && count2 < 2) {
+        long val = (long)entry;
+        TEST_ASSERT(val >= 20 && val < 25);
+        count2++;
+    }
+
+    /* Retarget back to first hashtable */
+    hashtableRetargetIterator(&iter, ht1);
+
+    /* Iterate partially through first hashtable again */
+    int count3 = 0;
+    while (hashtableNext(&iter, &entry) && count3 < 4) {
+        long val = (long)entry;
+        TEST_ASSERT(val >= 10 && val < 15);
+        count3++;
+    }
+
+    hashtableRelease(ht1);
+    hashtableRelease(ht2);
+
+    TEST_ASSERT(!hashtableNext(&iter, &entry));
+
+    hashtableCleanupIterator(&iter);
+
+    return 0;
+}
+ 
```

Comment:
There is a trailing space on this line which cases two CI errors:

* Clang-format https://github.com/valkey-io/valkey/actions/runs/19384385037/job/55469024225?pr=2807#step:5:17
* MacOS build: https://github.com/valkey-io/valkey/actions/runs/19384385019/job/55469024313?pr=2807#step:3:252

```suggestion

```